src/includes/class_reservationPermissions.php

```diff
@@ -1,76 +1,62 @@
 <?php
 class reservationPermissions {
 
-  public function getRecords($id = null){
-      try {
-          // call engine
-          $engine    = EngineAPI::singleton();
-          $localvars = localvars::getInstance();
-          $db        = db::get($localvars->get('dbConnectionName'));
-          $sql       = "SELECT * FROM `reservePermissions`";
-          $validate  = new validate;
+  private $localvars;
+  private $engine;
+  private $this->db;
+  private $validate;
+
+  function __construct() {
+    $this->engine    = EngineAPI::singleton();
+    $this->localvars = localvars::getInstance();
+    $this->db        = db::get($this->localvars->get('dbConnectionName'));
+    $this->validate  = new validate;
+  }
 
```

Comment:
Make these class variables. That way we only have to call those methods once, which is more efficient on the system and memory.